x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/reducer.ts

```diff
@@ -108,6 +108,28 @@ const getSourceIndexString = (state: State) => {
   return '';
 };
 
+const isSourceIndexNameValid = (
```

Comment:
nit: would be good to have a unit test for this function, but can be done in a follow up